sierra2mlir/src/utility.rs

```diff
@@ -199,55 +199,52 @@ impl<'ctx> Compiler<'ctx> {
         let entry_block = region
             .append_block(Block::new(&[(enum_type.get_type(), Location::unknown(&self.context))]));
 
-        let enum_value = entry_block.argument(0)?;
+        let enum_value = entry_block.argument(0)?.into();
 
         let function_type = create_fn_signature(&[enum_type.get_type()], &[]);
 
         if let SierraType::Enum {
-            ty,
+            ty: _,
             tag_type,
             storage_bytes_len: _,
-            storage_type: _,
+            storage_type,
             variants_types,
         } = enum_type
```

Comment:
When destructuring a struct-like construct without needing all its members you can avoid having to specify them all by doing something like this:
```rs
if let SierraType::Enum {
    tag_type,
    storage_type,
    variants_types,
    ..
} = enum_type
```